### PR TITLE
Expose Sigma rules overview and detail pages

### PR DESCRIPTION
**Pull request by jaegeral**
https://github.com/google/timesketch/issues/1850

1851_screen

Screenshot 2021-06-29 at 16 34 00

### CODE REVIEW

File: timesketch/frontend/src/components/AppNavbarSecondary.vue [hunk 1]
```diff
@@ -67,6 +67,12 @@ limitations under the License.
                   <span>Stories</span>
                 </router-link>
               </li>
+              <li v-bind:class="{ 'is-active': currentPage === 'sigma' }">
+                <router-link :to="{ name: 'SigmaOverview' }">
+                  <span class="icon is-small"><i class="fas fa-book" aria-hidden="true"></i></span>
```
Comment (berggren) [inline timesketch/frontend/src/components/AppNavbarSecondary.vue:72]: fa-book will collide with the Stories tab, right?

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 2]
```diff
@@ -0,0 +1,156 @@
+<!--
+Copyright 2020 Google Inc. All rights reserved.
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:2]:
```suggestion
Copyright 2021 Google Inc. All rights reserved.
```

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        sortable searchable>
+          {{ props.row.id }} 
+          <span class="icon is-small"
+                style="cursor:pointer;float:right;"
+                title="Copy value"
+                v-on:click="copyCode(props.row.id)">
+          <i class="fas fa-copy"></i></span>
+          <span class="icon is-small">
+                <router-link :to="{ name: 'SigmaContent', query: {ruleId: props.row.id }}"
+                   ><i class="fas fa-search">aaaaaaa</i></router-link></span>
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:47]: aaaaaaa? :)

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        </b-table-column>
+      </b-table>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      sigma_list: [],
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:109]: list of what? Sigma rules? then maybe call it `sigmaRuleList`? WDYT

Comment (berggren) [reply timesketch/frontend/src/components/Sigma/SigmaList.vue:109]: camel case for JS variables

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+          el.value = value;
+          document.body.appendChild(el);
+          el.select();
+          document.execCommand('copy');
+          this.$buefy.notification.open('Copied')
+          document.body.removeChild(el);
+    }
+  },
+  computed: {
+    sigmas() {
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:130]: sigmaRules?

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+<!-- CSS scoped to this component only -->
+<style scoped lang="scss">
+span {
+  background: red;
+}
+span:hover {
+  background: green;
+  visibility: visible;
+}
+</style>
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:156]: add newline

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    },
+    sketch() {
+      return this.$store.state.sketch
+    },
+    meta() {
+      return this.$store.state.meta
+    },
+  },
+  created() {
+    this.sigma_list =  this.$store.state.sigma_list
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:141]: This is redundant with the computed property `sigmas` above.
Just change the computed to do somehing like this:

```
sigmaRules() {
  return this.$store.state.sigma_list
}
```

Then you don't need to do this in the created() lifecycle hook.
See here for mote details on computed properties: https://vuejs.org/v2/guide/computed.html

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    this.sigma_list_count = this.$store.state.sigma_list_count
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:142]: What does this count do? Is it just a count of the list?
If that is the case, why not just do the count client side?

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+<!-- CSS scoped to this component only -->
+<style scoped lang="scss">
+span {
+  background: red;
+}
+span:hover {
+  background: green;
+  visibility: visible;
+}
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:155]: Left over from testing?

File: timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue [hunk 10]
```diff
@@ -0,0 +1,99 @@
+<!--
+Copyright 2020 Google Inc. All rights reserved.
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue:2]:
```suggestion
Copyright 2021 Google Inc. All rights reserved.
```

File: timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<template>
+  <div>
+      <ul class="content-list">
+        <li style="padding:10px;border-bottom:none;cursor:pointer;">
+          <strong style="color: var(--default-font-color)">{{ rule.id }}</strong>
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue:20]: Do we need this color style here?

File: timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+          <br />
+          {{ ruleId }}
+          <div v-for="(value, name) in rule" v-bind:key="name">
+            {{ name }}: {{ value }}
+        </div>
+        </li>
+      </ul>
+
+  <!-- jaegeral: This is draft currently -->
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue:29]: is it still draft? :)

File: timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+    meta() {
+      return this.$store.state.meta
+    },
+  },
+  created() {
+    // TOTO (jaegeral): change that to use the rule from the already stored one
+    ApiClient.getSigmaResource(this.$route.query.ruleId)
+      .then(response => {
+        this.rule = response.data['objects'][0]
+        console.log(this.rule)
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue:72]: remove console.log

File: timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+    },
+    submitForm: function() {
+      ApiClient.getSigmaByText(this.SigmaText)
+        .then(response => {
+          let newView = response.data.objects[0]
+          console.log(newView)
+          //this.$emit('setActiveView', newView)
+          //this.$store.state.meta.views.push(newView)
+          //this.clearFormData()
+          //this.$router.push({ name: 'Explore', query: { view: newView.id } })
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue:90]: remove?

File: timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+</template>
+
+<script>
+import ApiClient from '../../utils/RestApiClient'
+
+export default {
+  data() {
+    return {
+      rule: [],
+      SigmaText : '',
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue:53]: camelcase: `sigmaText`

File: timesketch/frontend/src/views/Sigma.vue [hunk 16]
```diff
@@ -0,0 +1,18 @@
+<!--
+Copyright 2020 Google Inc. All rights reserved.
```
Comment (berggren) [inline timesketch/frontend/src/views/Sigma.vue:2]:
```suggestion
Copyright 2021 Google Inc. All rights reserved.
```

File: timesketch/frontend/src/views/SigmaContent.vue [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+        <section class="section">
+
+    <ts-navbar-main>
+      <template v-slot:left>
+        {{ sketch.name }}
+      </template>
+    </ts-navbar-main>
+
+    <ts-navbar-secondary currentAppContext="sketch" currentPage="sigma"></ts-navbar-secondary>
+      <div class="container is-fluid">
```
Comment (berggren) [inline timesketch/frontend/src/views/SigmaContent.vue:27]: weird indentation here. I'll show how to set up auto formatting that will help here :)
Install `prettier` in vscode, and config it to auto format on save:
https://scottsauber.com/2017/06/10/prettier-format-on-save-never-worry-about-formatting-javascript-again/

File: timesketch/frontend/src/views/SigmaOverview.vue [hunk 18]
```diff
@@ -0,0 +1,51 @@
+<!--
+Copyright 2020 Google Inc. All rights reserved.
```
Comment (berggren) [inline timesketch/frontend/src/views/SigmaOverview.vue:2]:
```suggestion
Copyright 2021 Google Inc. All rights reserved.
```

**Review by berggren — CHANGES_REQUESTED** [review]
First round done

File: timesketch/frontend/src/components/AppNavbarSecondary.vue (hunk 1 above)
Comment (jaegeral) [reply timesketch/frontend/src/components/AppNavbarSecondary.vue:72]: yes, let me use another one

File: timesketch/frontend/src/components/Sigma/SigmaList.vue (hunk 9 above)
Comment (jaegeral) [reply timesketch/frontend/src/components/Sigma/SigmaList.vue:155]: yes

File: timesketch/frontend/src/components/Sigma/SigmaList.vue (hunk 4 above)
Comment (jaegeral) [reply timesketch/frontend/src/components/Sigma/SigmaList.vue]: got rid of it as it was no longer needed

File: timesketch/frontend/src/components/Sigma/SigmaList.vue (hunk 8 above)
Comment (jaegeral) [reply timesketch/frontend/src/components/Sigma/SigmaList.vue]: Yes it is the count of the list but is given by the API as a value, so it does not hurt to use it or?

File: timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue (hunk 11 above)
Comment (jaegeral) [reply timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue:20]: no we do not

File: timesketch/frontend/src/views/SigmaContent.vue (hunk 17 above)
Comment (jaegeral) [reply timesketch/frontend/src/views/SigmaContent.vue:27]: TIL, thx!

**Comment by jaegeral** [comment]
@berggren PTAL (besides the copy button, that is subject to change after we got the other PR in I will use the same mechanism here

File: timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+limitations under the License.
+-->
+<template>
+  <div>
+    <strong>{{ rule.id }}</strong>
+    <ul class="content-list">
+      <li v-for="(value, name) in rule" v-bind:key="name" style="padding:10px;border-bottom:none;cursor:pointer;">
+        <div>
+          <b>{{ name }}</b
+          >: <code>{{ value }}</code>
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaRuleDetail.vue:23]: Can we make this a table instead? That will fit into the UI a bit better seeing that the event details is also a table.

File: timesketch/frontend/src/components/Sigma/SigmaList.vue [hunk 20]
```diff
@@ -0,0 +1,129 @@
...
+      pagination-position="bottom"
+      default-sort-direction="desc"
+      sort-icon="arrow-down"
+      sort-icon-size="is-small"
+      icon-pack="fas"
+      icon-prev="chevron-left"
+      icon-next="chevron-right"
+      default-sort="title"
+    >
+      <b-table-column field="id" label="Rule ID" v-slot="props" sortable searchable>
```
Comment (berggren) [inline timesketch/frontend/src/components/Sigma/SigmaList.vue:34]: I'm about to mock my UI idea here. I'll update tomorrow.
I'm windering if we should just have one button to take the user to the explore view - with the generated query as the parameter?

**Review by berggren — CHANGES_REQUESTED** [review]
Looking good! Just a few small things, and some UI related thoughts. Lets chat tomorrow on how to proceed.

File: timesketch/frontend/src/components/Sigma/SigmaList.vue (hunk 20 above)
Comment (jaegeral) [reply timesketch/frontend/src/components/Sigma/SigmaList.vue:34]: yes that is super easy to do and is already in the overview table, I just did not implement it here to wait on the brainstorming

**Comment by jaegeral** [comment]
Some updates. I have not yet made it disappear in the Table, as @binglot told me he is working on a general hover context menu which we might want to use here instead. Will push it over to Bart for a review from his side.

Screenshot 2021-07-07 at 16 50 59
Screenshot 2021-07-07 at 16 29 14

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**